src/compatibility/fvSolution.cpp

```diff
@@ -96,32 +111,46 @@ void updateCriteria(NeoN::Dictionary& solverDict)
         solverDict.insert("criteria", NeoN::Dictionary());
     }
 
-
-    NeoN::Dictionary& criteriaDict = solverDict.subDict("criteria");
-
     // set default max iteration count
-    criteriaDict.insert("iteration", 1000);
+    {
+        NeoN::Dictionary& criteriaDict = solverDict.subDict("criteria");
+        criteriaDict.insert("iteration", 1000);
+    }
 
     // Set default values for relative residual norm and iteration count
     if (solverDict.contains("relTol"))
     {
+        NeoN::Dictionary& criteriaDict = solverDict.subDict("criteria");
         criteriaDict.insert("relative_residual_norm", solverDict.get<NeoN::scalar>("relTol"));
+        solverDict.remove("relTol");
     }
     if (solverDict.contains("maxIter"))
     {
+        NeoN::Dictionary& criteriaDict = solverDict.subDict("criteria");
         criteriaDict.insert("iteration", solverDict.get<NeoN::label>("maxIter"));
+        solverDict.remove("maxIter");
     }
     if (solverDict.contains("tolerance"))
     {
+        NeoN::Dictionary& criteriaDict = solverDict.subDict("criteria");
         criteriaDict.insert("absolute_residual_norm", solverDict.get<NeoN::scalar>("tolerance"));
+        solverDict.remove("tolerance");
     }
+
+    NeoN::Dictionary& criteriaDict = solverDict.subDict("criteria");
+    std::cout << __FILE__ << ":\n\tStopping criteria: " << criteriaDict << "\n";
 }
 
 
 NeoN::Dictionary mapFvSolution(const NeoN::Dictionary& solverDict)
 {
     NeoN::Dictionary modSolverDict = solverDict;
 
+    if (solverDict.contains("configFile")) return solverDict;
```

Comment:
```suggestion
    if (modSolverDict.contains("configFile")) return modSolverDict;
```
To avoid confusion from aliasing the input dictionray